src/screens/GroupSetting/hooks/useGroupSetting.js

```diff
@@ -36,53 +50,74 @@ const useGroupSetting = ({navigation, route}) => {
     }
   }, [route]);
 
-  const submitData = async (withNavigation = true, withLoading = true) => {
-    let changeImageUrl = '';
-    if (changeImage) {
-      if (withLoading) setIsLoading(true);
-      try {
-        const res = await uploadFile(`data:image/jpeg;base64,${base64Profile}`);
-        changeImageUrl = res.data.url;
-      } catch (e) {
-        setIsLoading(false);
-        return;
-      }
+  React.useEffect(() => {
+    if (selectedChannel) {
+      setUrlImage(selectedChannel?.channelPicture);
     }
-
-    if (changeName || changeImage) {
-      if (withLoading) setIsLoading(true);
-      let dataEdit = {
-        name: groupName
-        // ...(changeImage && {image: base64Profile}),
+  }, [selectedChannel]);
+  const submitData = async (withNavigation = true) => {
+    if (!changeImage && !changeName) return navigation.goBack();
+    try {
+      setLoadingUpdate(true);
+      let body = {
+        channel_id: route?.params?.id,
+        channel_name: groupName,
+        channel_image: urlImage
       };
-      if (changeName) {
-        dataEdit = {...dataEdit, isEditName: true};
+      if (urlImage && !isValidUrl(urlImage)) {
+        try {
+          const res = await uploadPhotoImage(urlImage);
+          if (res) {
+            body = {...body, channel_image: res.data.url};
+          }
+        } catch (e) {
+          console.log(e, 'error upload');
+        }
       }
-      if (changeImage) {
-        dataEdit.image = changeImageUrl;
-      } else if (channel?.data?.image) {
-        dataEdit.image = channel?.data?.image;
+      const response = await SignedMessageRepo.editChannel(body);
+      let newChannel = response.data?.channel;
+      newChannel = {
+        ...chat.selectedChannel,
+        ...newChannel,
+        channelType: handleChannelType(newChannel?.channelType),
+        channelPicture: body?.channel_image
+      };
+      ChannelList.updateChannelInfo(
+        localDb,
+        body.channel_id,
+        body.channel_name,
+        body.channel_image
+      );
+      if (newChannel) {
+        setChat({
+          ...chat,
+          selectedChannel: newChannel
+        });
       }
-      console.log({dataEdit}, 'nusu');
-      updateDataEdit(dataEdit, withNavigation);
-    } else if (withNavigation) navigation.goBack();
-  };
+      refresh('channelInfo');
+      refresh('chat');
 
-  const updateDataEdit = async (dataEdit, withNavigation) => {
-    try {
-      await channel.update(dataEdit);
-      if (withNavigation) navigation.navigate('ChatDetailPage');
-    } catch (e) {
-      if (__DEV__) {
-        console.log(`error : ${e}`);
+      refresh('channelList');
+
+      if (withNavigation) {
+        navigation.goBack();
       }
+      setLoadingUpdate(false);
+    } catch (e) {
+      setLoadingUpdate(false);
       SimpleToast.show(StringConstant.groupSettingUpdateFailed);
     }
-    setIsLoading(false);
+  };
+
+  const handleChannelType = (channelType) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential issue with error handling in the `submitData` function. When an error occurs during the image upload process, the function logs the error but continues to execute the rest of the code. This could lead to unexpected behavior if the image upload is critical for the subsequent steps. Consider adding error handling or control flow to stop execution when the image upload fails.

```diff
      if (urlImage && !isValidUrl(urlImage)) {
        try {
          const res = await uploadPhotoImage(urlImage);
          if (res) {
            body = {...body, channel_image: res.data.url};
          }
        } catch (e) {
          console.log(e, 'error upload');
+          throw e;
        }
       }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->